qiskit_ibm_runtime/noise_learner/noise_learner.py

```diff
@@ -0,0 +1,268 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Noise learner program."""
+
+from __future__ import annotations
+
+from copy import deepcopy
+from dataclasses import asdict, fields, replace
+from typing import Any, Dict, Iterable, Optional, Union
+import logging
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.providers import BackendV2
+from qiskit.primitives.containers import EstimatorPubLike
+from qiskit.primitives.containers.estimator_pub import EstimatorPub
+
+from ..base_primitive import _get_mode_service_backend
+from ..constants import DEFAULT_DECODERS
+from ..runtime_job_v2 import RuntimeJobV2
+from ..ibm_backend import IBMBackend
+from ..options.estimator_options import EstimatorOptions
+from ..options.noise_learner_options import NoiseLearnerOptions
+from ..options.utils import remove_dict_unset_values, remove_empty_dict
+from ..utils import validate_isa_circuits
+from ..utils.utils import is_simulator
+
+from ..fake_provider.local_service import QiskitRuntimeLocalService
+from ..qiskit_runtime_service import QiskitRuntimeService
+
+
+# pylint: disable=unused-import,cyclic-import
+from ..session import Session
+from ..batch import Batch
+
+logger = logging.getLogger(__name__)
+
+
+class NoiseLearner:
+    """Class for executing noise learning experiments.
+
```

Comment:
If we plan to release this as experimental, then we should have add a warning here.

Reply:
There is a note:
```
    .. note::
        Currently, the :class:`.NoiseLearner` is released an experimental feature.
        As such, it is subject to change without notification and its stability is not
        guaranteed.
```
Would you feel safer with a warning, in that it is more explicit?

Reply:
Okay, moved it :) 

```
class NoiseLearner:
    """Class for executing noise learning experiments.

    .. note::
        Currently, the :class:`.NoiseLearner` is released an experimental feature.
        As such, it is subject to change without notification and its stability is not
        guaranteed.
```